blogposts/2023-02-10-adding-snapshot-testing-to-rnw-apps.md

```diff
@@ -0,0 +1,93 @@
+
+Here is an example of the code added for the `react-native-permissions` module:
+```js
+jest.mock('react-native-permissions', () =>
+require('react-native-permissions/mock'),
+);
+```
+If the community module does not have a mock.js file, you'll have to mock the undefined function yourself. There are a couple ways to sleuth out how to correctly mock: 
+	1. Check the community module source. If they have an example app with Jest testing, what do they do in their `jest-setup.js` file?
+	2. You can also check postings from the community on how they mocked particular methods for a given module.
+	3. React Native Gallery has a wide set of community modules it has tests for. Check out our [`jest-setup.js`](https://github.com/microsoft/react-native-gallery/blob/main/jest-setup.js) to see how we mocked needed modules. 
+
+Here is React Native Gallery's Jest customization within [`package.json`](https://github.com/microsoft/react-native-gallery/blob/main/package.json):
+```json
+"jest": {
+    "preset": "react-native",
+    "modulePathIgnorePatterns": [
+      "<rootDir>/lib/"
+    ],
+    "transformIgnorePatterns": [
+      "node_modules/(?!(react-native|react-native-xaml|@react-native|react-native-windows|react-native-config|@react-native-community|react-native-print|react-native-webview|react-native-windows-hello|react-native-permissions|react-native-tts|react-native-gesture-handler)/)"
+    ],
+    "setupFiles": [
+      "<rootDir>/jest-setup.js"
+    ]
+  },
+```
+
+Once you have all your tests running and passing, you might see that the Jest run is still exiting with exit code 1. This occurred in React Native Gallery. After some searching, I discovered it was because some of the components our tests were rendering contained asynchronous calls and updates. This results in Jest throwing errors because it cannot ensure that all tasks such as renders, user events, or data fetching were completed and applied before the snapshot match assertion was made. To resolve this error, you'll need to wrap the render of the component you wish to snapshot in React's `act()` function. Then, following the `act()` call, make your snapshot assertion. 
```

Comment:
> Once you have all your tests running and passing, you might see that the Jest run is still exiting with exit code 1.

I'm confused: are the tests running and passing, or is Jest existing with code 1? If it's an actual scenario, can we add a few more details to the phrase to make it clearer (ex. "all your individual tests")

Reply:
Sure I can add more clarity here!